Current Build doesn't work

I attempted to spin up a new gbl_admin app using the template and I get errors that prevent it from working. Looks like something in Blacklight has issues with Pagy:

```
1 installed gem you directly depend on is looking for funding.
  Run `bundle fund` for details
    generate    blacklight:install
       rails    generate blacklight:install --devise
/Users/mac/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/bundler/gems/geoblacklight_admin-6a7df5ae343f/app/controllers/concerns/blacklight/catalog.rb:4:in '<module:Catalog>': uninitialized constant Blacklight::Catalog::Pagy (NameError)

  include Pagy::Backend
          ^^^^
	from /Users/me/.local/share/mise/installs/ruby/3.4.2/lib/ruby/gems/3.4.0/bundler/gems/geoblacklight_admin-6a7df5ae343f/app/controllers/concerns/blacklight/catalog.rb:3:in '<main>'
```
Using:
Rails v7.2.2.1
Ruby v3.4.2

I did also hit an issue with compiling that I solved I think by installing gcc and some other things but maybe it is related?

```

Installing geoblacklight 4.4.2
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/Users/me/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/sassc-2.4.0/ext
/Users/me/.rbenv/versions/3.4.2/bin/ruby extconf.rb
creating Makefile

current directory:
/Users/me/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/sassc-2.4.0/ext
make DESTDIR\= sitearchdir\=./.gem.20250416-57175-6mqx50
sitelibdir\=./.gem.20250416-57175-6mqx50 clean

current directory:
/Users/me/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/sassc-2.4.0/ext
make DESTDIR\= sitearchdir\=./.gem.20250416-57175-6mqx50
sitelibdir\=./.gem.20250416-57175-6mqx50
compiling ./libsass/src/ast.cpp
make: *** [ast.o] Error 1

make failed, exit code 2
```

Hit this in my local mac environment.

--- a/lib/generators/geoblacklight_admin/templates/.solr_wrapper.yml
+++ b/lib/generators/geoblacklight_admin/templates/.solr_wrapper.yml
@@ -1,4 +1,3 @@
-url: https://archive.apache.org/dist/lucene/solr/8.11.4/solr-8.11.4.tgz
 validate: false
 instance_dir: tmp/solr
 download_dir: tmp/solr

--- a/template.rb
+++ b/template.rb
@@ -29,26 +29,30 @@ def inject_geoblacklight_admin_dependencies
   begin
     gemspec_content = URI.open(gemspec_url).read
     
-    # Extract dependencies using regex instead of evaluating the gemspec
-    dependencies = gemspec_content.scan(/^\s*s\.add_dependency\s*["']([^"']+)["'](,\s*["'][^"']+["'])?/)
+    # Match lines like: s.add_dependency "gem_name", "~> 1.0"
+    dependency_pattern = /^\s*s\.add_dependency\s*["']([^"']+)["'](,\s*["'][^"']+["'])?/
     
-    dependencies.each do |dep|
-      name = dep[0]
-      version = dep[1]&.gsub(/[",\s]/, '') # Clean up version constraint
-      
-      # Skip if the dependency is already declared
-      next if gem_already_declared?(name)
-      
-      if version
-        gem name, version
-      else
-        gem name
+    gemspec_content.each_line do |line|
+      if match = line.match(dependency_pattern)
+        name = match[1]
+        version = match[2]&.gsub(/[",\s]/, '') # Clean up version constraint
+        
+        # Skip if the dependency is already declared
+        next if gem_already_declared?(name)
+        
+        if version
+          say "Adding dependency: #{name} (#{version})", :green
+          gem name, version
+        else
+          say "Adding dependency: #{name}", :green
+          gem name
+        end
       end
     end
   rescue OpenURI::HTTPError => e
-    say "Warning: Could not fetch gemspec from #{gemspec_url}: #{e.message}", :yellow
+    say "Warning: Could not fetch gemspec from #{gemspec_url}: #{e.message}", :red
   rescue StandardError => e
-    say "Warning: Error processing gemspec: #{e.message}", :yellow
+    say "Warning: Error processing gemspec: #{e.message}", :red
   end
 end
 

--- a/template.rb
+++ b/template.rb
@@ -20,48 +20,39 @@ gem "blacklight", ">= 7.0", "< 8.0"
 gem "geoblacklight", ">= 4.0"
 gem "geoblacklight_admin", git: "git@github.com:geobtaa/geoblacklight_admin.git", branch: "develop"
 
-# Inject geoblacklight_admin dependencies
-def inject_geoblacklight_admin_dependencies
-  require 'open-uri'
-  
-  gemspec_url = "https://raw.githubusercontent.com/geobtaa/geoblacklight_admin/develop/geoblacklight_admin.gemspec"
-  
-  begin
-    gemspec_content = URI.open(gemspec_url).read
-    
-    # Match lines like: s.add_dependency "gem_name", "~> 1.0"
-    dependency_pattern = /^\s*s\.add_dependency\s*["']([^"']+)["'](,\s*["'][^"']+["'])?/
-    
-    gemspec_content.each_line do |line|
-      if match = line.match(dependency_pattern)
-        name = match[1]
-        version = match[2]&.gsub(/[",\s]/, '') # Clean up version constraint
-        
-        # Skip if the dependency is already declared
-        next if gem_already_declared?(name)
-        
-        if version
-          say "Adding dependency: #{name} (#{version})", :green
-          gem name, version
-        else
-          say "Adding dependency: #{name}", :green
-          gem name
-        end
-      end
-    end
-  rescue OpenURI::HTTPError => e
-    say "Warning: Could not fetch gemspec from #{gemspec_url}: #{e.message}", :red
-  rescue StandardError => e
-    say "Warning: Error processing gemspec: #{e.message}", :red
-  end
-end
-
-def gem_already_declared?(name)
-  # Check if gem is already declared in template
-  File.readlines(__FILE__).any? { |line| line.match?(/^\s*gem\s+["']#{name}["']/) }
-end
-
-inject_geoblacklight_admin_dependencies
+# GBL‡ADMIN
+# Add geoblacklight_admin runtime dependencies
+gem "active_storage_validations", "~> 1.0"
+gem "amazing_print"
+gem "blacklight", "~> 7.0"
+gem "blacklight_advanced_search"
+gem "blacklight_range_limit"
+gem "bootstrap", "~> 4.0"
+gem "chosen-rails", "~> 1.10"
+gem "cocoon", "~> 1.2"
+gem "config", "~> 4.0"
+gem "devise", "~> 4.7"
+gem "devise-bootstrap-views", "~> 1.0"
+gem "dotenv-rails", "~> 2.8"
+gem "geoblacklight", "~> 4.0"
+gem "haml", "~> 5.2"
+gem "httparty", "~> 0.21"
+gem "inline_svg", "~> 1.9"
+gem "jquery-rails", "~> 4.4"
+gem "kithe", "~> 2.0"
+gem "mutex_m", "~> 0.2.0"
+gem "noticed", "~> 1.6"
+gem "pagy", "~> 9.0"
+gem "paper_trail", "~> 15.0"
+gem "pg", "~> 1.4"
+gem "qa", "~> 5.0"
+gem "ruby-progressbar"
+gem "simple_form", "~> 5.0"
+gem "sprockets", "~> 3.0"
+gem "statesman", "~> 12.0"
+gem "vite_rails", "~> 3.0"
+gem "vite_ruby", ">= 3.5"
+gem "zeitwerk", "~> 2.6"
 
 run "bundle install"
 

--- a/template.rb
+++ b/template.rb
@@ -14,45 +14,22 @@
 # $> bundle exec rake geoblacklight:server
 #
 
-# Core dependencies
 gem "devise"
 gem "blacklight", ">= 7.0", "< 8.0"
 gem "geoblacklight", ">= 4.0"
 gem "geoblacklight_admin", git: "git@github.com:geobtaa/geoblacklight_admin.git", branch: "develop"
 
 # GBL‡ADMIN
-# Add geoblacklight_admin runtime dependencies
-gem "active_storage_validations", "~> 1.0"
-gem "amazing_print"
-gem "blacklight", "~> 7.0"
+gem "active_storage_validations"
+gem "awesome_print"
 gem "blacklight_advanced_search"
-gem "blacklight_range_limit"
-gem "bootstrap", "~> 4.0"
-gem "chosen-rails", "~> 1.10"
-gem "cocoon", "~> 1.2"
-gem "config", "~> 4.0"
-gem "devise", "~> 4.7"
-gem "devise-bootstrap-views", "~> 1.0"
-gem "dotenv-rails", "~> 2.8"
-gem "geoblacklight", "~> 4.0"
-gem "haml", "~> 5.2"
-gem "httparty", "~> 0.21"
-gem "inline_svg", "~> 1.9"
-gem "jquery-rails", "~> 4.4"
+gem "dotenv-rails"
+gem "haml"
+gem "inline_svg"
 gem "kithe", "~> 2.0"
-gem "mutex_m", "~> 0.2.0"
-gem "noticed", "~> 1.6"
+gem "noticed"
+gem "paper_trail"
 gem "pagy", "~> 9.0"
-gem "paper_trail", "~> 15.0"
-gem "pg", "~> 1.4"
-gem "qa", "~> 5.0"
-gem "ruby-progressbar"
-gem "simple_form", "~> 5.0"
-gem "sprockets", "~> 3.0"
-gem "statesman", "~> 12.0"
-gem "vite_rails", "~> 3.0"
-gem "vite_ruby", ">= 3.5"
-gem "zeitwerk", "~> 2.6"
 
 run "bundle install"
 

--- a/template.rb
+++ b/template.rb
@@ -22,35 +22,33 @@ gem "geoblacklight_admin", git: "git@github.com:geobtaa/geoblacklight_admin.git"
 
 # Inject geoblacklight_admin dependencies
 def inject_geoblacklight_admin_dependencies
-  require 'bundler'
   require 'open-uri'
   
   gemspec_url = "https://raw.githubusercontent.com/geobtaa/geoblacklight_admin/develop/geoblacklight_admin.gemspec"
   
   begin
     gemspec_content = URI.open(gemspec_url).read
-    # Create a temporary file to evaluate the gemspec
-    tempfile = Tempfile.new(['geoblacklight_admin', '.gemspec'])
-    tempfile.write(gemspec_content)
-    tempfile.close
     
-    gemspec = Bundler.load_gemspec(tempfile.path)
-    gemspec.runtime_dependencies.each do |dep|
-      # Skip if the dependency is already declared
-      next if gem_already_declared?(dep.name)
+    # Extract dependencies using regex instead of evaluating the gemspec
+    dependencies = gemspec_content.scan(/^\s*s\.add_dependency\s*["']([^"']+)["'](,\s*["'][^"']+["'])?/)
+    
+    dependencies.each do |dep|
+      name = dep[0]
+      version = dep[1]&.gsub(/[",\s]/, '') # Clean up version constraint
       
-      if dep.requirements_list.any?
-        gem dep.name, *dep.requirements_list
+      # Skip if the dependency is already declared
+      next if gem_already_declared?(name)
+      
+      if version
+        gem name, version
       else
-        gem dep.name
+        gem name
       end
     end
   rescue OpenURI::HTTPError => e
     say "Warning: Could not fetch gemspec from #{gemspec_url}: #{e.message}", :yellow
   rescue StandardError => e
     say "Warning: Error processing gemspec: #{e.message}", :yellow
-  ensure
-    tempfile&.unlink
   end
 end
 

--- a/template.rb
+++ b/template.rb
@@ -20,13 +20,21 @@ gem "blacklight", ">= 7.0", "< 8.0"
 gem "geoblacklight", ">= 4.0"
 gem "geoblacklight_admin", git: "git@github.com:geobtaa/geoblacklight_admin.git", branch: "develop"
 
-# GBL‡ADMIN
 # Inject geoblacklight_admin dependencies
 def inject_geoblacklight_admin_dependencies
   require 'bundler'
-  gemspec_path = File.join(File.dirname(__FILE__), 'geoblacklight_admin.gemspec')
-  if File.exist?(gemspec_path)
-    gemspec = Bundler.load_gemspec(gemspec_path)
+  require 'open-uri'
+  
+  gemspec_url = "https://raw.githubusercontent.com/geobtaa/geoblacklight_admin/develop/geoblacklight_admin.gemspec"
+  
+  begin
+    gemspec_content = URI.open(gemspec_url).read
+    # Create a temporary file to evaluate the gemspec
+    tempfile = Tempfile.new(['geoblacklight_admin', '.gemspec'])
+    tempfile.write(gemspec_content)
+    tempfile.close
+    
+    gemspec = Bundler.load_gemspec(tempfile.path)
     gemspec.runtime_dependencies.each do |dep|
       # Skip if the dependency is already declared
       next if gem_already_declared?(dep.name)
@@ -37,8 +45,12 @@ def inject_geoblacklight_admin_dependencies
         gem dep.name
       end
     end
-  else
-    say "Warning: Could not find geoblacklight_admin.gemspec", :yellow
+  rescue OpenURI::HTTPError => e
+    say "Warning: Could not fetch gemspec from #{gemspec_url}: #{e.message}", :yellow
+  rescue StandardError => e
+    say "Warning: Error processing gemspec: #{e.message}", :yellow
+  ensure
+    tempfile&.unlink
   end
 end
 

--- a/template.rb
+++ b/template.rb
@@ -29,6 +29,7 @@ gem "inline_svg"
 gem "kithe", "~> 2.0"
 gem "noticed"
 gem "paper_trail"
+gem "pagy", "~> 9.0"
 
 run "bundle install"
 

--- a/template.rb
+++ b/template.rb
@@ -14,22 +14,40 @@
 # $> bundle exec rake geoblacklight:server
 #
 
+# Core dependencies
 gem "devise"
 gem "blacklight", ">= 7.0", "< 8.0"
 gem "geoblacklight", ">= 4.0"
 gem "geoblacklight_admin", git: "git@github.com:geobtaa/geoblacklight_admin.git", branch: "develop"
 
 # GBL‡ADMIN
-gem "active_storage_validations"
-gem "awesome_print"
-gem "blacklight_advanced_search"
-gem "dotenv-rails"
-gem "haml"
-gem "inline_svg"
-gem "kithe", "~> 2.0"
-gem "noticed"
-gem "paper_trail"
-gem "pagy", "~> 9.0"
+# Inject geoblacklight_admin dependencies
+def inject_geoblacklight_admin_dependencies
+  require 'bundler'
+  gemspec_path = File.join(File.dirname(__FILE__), 'geoblacklight_admin.gemspec')
+  if File.exist?(gemspec_path)
+    gemspec = Bundler.load_gemspec(gemspec_path)
+    gemspec.runtime_dependencies.each do |dep|
+      # Skip if the dependency is already declared
+      next if gem_already_declared?(dep.name)
+      
+      if dep.requirements_list.any?
+        gem dep.name, *dep.requirements_list
+      else
+        gem dep.name
+      end
+    end
+  else
+    say "Warning: Could not find geoblacklight_admin.gemspec", :yellow
+  end
+end
+
+def gem_already_declared?(name)
+  # Check if gem is already declared in template
+  File.readlines(__FILE__).any? { |line| line.match?(/^\s*gem\s+["']#{name}["']/) }
+end
+
+inject_geoblacklight_admin_dependencies
 
 run "bundle install"
 

